Tests/MongoSwiftSyncTests/MongoDatabaseTests.swift

```diff
@@ -220,6 +220,43 @@ final class MongoDatabaseTests: MongoSwiftTestCase {
         }
     }
 
+    func testListCollectionNamesAuthorizedCollections() throws {
+        // check if collection authorization is passed and received correctly
+        let client = try MongoClient.makeTestClient()
+        let monitor = client.addCommandMonitor()
+        let db = client.db(Self.testDatabase)
+        try db.drop()
+
+        _ = try db.createCollection("foo")
+        _ = try db.createCollection("bar")
+        _ = try db.createCollection("baz")
+
+        try monitor.captureEvents {
+            var opts = ListCollectionsOptions(authorizedCollections: true)
+            _ = try db.listCollectionNames(nil, options: opts, session: nil)
```

Comment:
minor style comment: since `filter` (the first parameter) and `session` already have default values of `nil` for `listCollections` and `listCollectionNames`, we can just omit them and only pass the `options: opts` part.